### PR TITLE
Add support for SSH Agent

### PR DESCRIPTION
**Pull request by paultag**
Adding a key this way requires no changes to .ssh/config, only a running ssh-agent, which reduces a lot of the client setup burden.

### CODE REVIEW

File: cmd/hallow-cli/main.go [hunk 1]
```diff
@@ -16,6 +17,44 @@ func hallowClientFromCLI(c *cli.Context) client.Client {
 	return client.New(session.New(), http.DefaultClient, c.String("endpoint"))
 }
 
+//
+func keyTypeFromCLI(c *cli.Context) (client.KeyType, error) {
+	switch c.String("key-type") {
```
[the remark this answers is not in the document]
Comment (paultag) [reply cmd/hallow-cli/main.go]: @alex I was attempting to match ssh's `-t` and `-b` args. Happy to do it the other way. Thoughts?

File: client/generate.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		case KeyTypeECDSAP224:
+			curve = elliptic.P224()
+		case KeyTypeECDSAP256:
+			curve = elliptic.P256()
+		case KeyTypeECDSAP384:
+			curve = elliptic.P384()
+		case KeyTypeECDSAP521:
+			curve = elliptic.P521()
+		default:
+			return nil, nil, fmt.Errorf("hallow/client: unknown ecdsa curve")
```
[the remark this answers is not in the document]
Comment (paultag) [reply client/generate.go]: True, I was setting that up to not make mistakes when adding new curves, but I guess it should be obvious to the person doing the PR.

File: cmd/hallow-cli/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (alex) [reply cmd/hallow-cli/main.go]: Hmmm, I'm on the fence. I like my way better (:P), I think it's behavior is more obvious and easier to read/review, but consistency with `ssh-keygen` does make sense.

[the remark this answers is not in the document]
Comment (paultag) [reply cmd/hallow-cli/main.go]: I actually don't have an opinion - let's change it. I don't like `bits` for the curve.

[the remark this answers is not in the document]
Comment (paultag) [reply cmd/hallow-cli/main.go]: Fixed!

**Pull request merged**